packages/list-store/src/use-searchable-facet.tsx

```diff
@@ -0,0 +1,227 @@
+      })),
+    setSortBy: (sortBy: FacetSortBy) =>
+      set(() => ({
+        sortBy,
+      })),
+    setFilters: (filters: Filter[]) =>
+      set(() => ({
+        filters,
+      })),
+  }));
+};
+
+type FacetProviderProps = React.PropsWithChildren<Partial<FacetProps>>;
+
+export const FacetContext = createContext<FacetStore | null>(null);
+
+export function FacetProvider({children, ...props}: FacetProviderProps) {
+  const storeRef = useRef<FacetStore>();
+  if (!storeRef.current) {
+    storeRef.current = createFacetStore(props);
+  }
+  return (
+    <FacetContext.Provider value={storeRef.current}>
+      {children}
+    </FacetContext.Provider>
+  );
+}
+
+interface GetLetterCategoriesProps {
+  filters: SearchableFilter[];
+  letterCategory?: string;
+  searchValue?: string;
+}
+
+export function getLetterCategories({
+  filters,
+  letterCategory,
+  searchValue,
+}: GetLetterCategoriesProps) {
+  const filtersFilteredBySearchValue = filterBySearchValue(
+    filters,
+    searchValue
+  );
+
+  return [
+    ...new Set([
+      // Always include the letter category of the current filter
+      ...(letterCategory ? [letterCategory] : []),
+      ...filtersFilteredBySearchValue
+        .map(filter => filter.letterCategory)
+        .filter(category => !!category),
+    ]),
+  ].sort();
+}
+
+export function extendFiltersWithLetterCategory(filters?: Filter[]) {
+  if (!filters?.length) {
+    return [];
+  }
+  return filters.map(filter => {
+    let letterCategory = '';
+    const name = filter.name?.toString();
+    if (name) {
+      const firstLetter = name.toUpperCase().replace(/[^0-9A-Z]/g, '')[0];
+      if (/^\d+$/.test(firstLetter)) {
+        letterCategory = '[0-9]';
+      } else {
+        letterCategory = firstLetter;
+      }
+    }
+    return {
+      ...filter,
+      name: name || '',
+      letterCategory,
+    };
+  });
+}
+
+interface GetFilteredFiltersProps {
+  filtersWithLetterCategory: SearchableFilter[];
+  searchValue: string;
+  letterCategory: string;
+  sortBy: FacetSortBy;
+}
+
+function filterBySearchValue(
+  filters: SearchableFilter[],
+  searchValue?: string
+) {
+  if (searchValue) {
+    return filters.filter(
+      filter => filter.name?.toLowerCase().includes(searchValue.toLowerCase())
+    );
+  } else return filters;
+}
+
+function filterByLetterCategory(
+  filters: SearchableFilter[],
+  letterCategory: string
+) {
+  if (letterCategory) {
+    return filters.filter(filter => filter.letterCategory === letterCategory);
+  } else return filters;
+}
+
+export function getFilteredFilters({
+  filtersWithLetterCategory,
+  searchValue,
+  letterCategory,
+  sortBy,
+}: GetFilteredFiltersProps) {
+  const filtersFilteredBySearchValue = filterBySearchValue(
+    filtersWithLetterCategory,
+    searchValue
+  );
+
+  const filtersFilteredByLetterCategory = filterByLetterCategory(
+    filtersFilteredBySearchValue,
+    letterCategory
+  );
+
+  if (sortBy === 'alphabetical') {
```

Comment:
Is this `alphabetical` the same as the one in `FacetSortBy`? If so, use the latter?

Reply:
Do you mean `if (sortBy === FacetSortBy.alphabetical) {`?

For that, `FacetSortBy` needs to be an `enum`. This was a literal type, not an enum. I don't really like typescript enums for different reasons. But we are probably too far in this project to start this discussion. So, I will change it to an enum.

Reply:
Ok! But do you still want to do a `if (sortBy === FacetSortBy.alphabetical)`?

Reply:
Sorry, I thought I had done that!